### DISCUSSION TITLE
Setting up Perl PATH for executables

### DISCUSSION THREAD

**Discussion by dhorne** [Homebrew/discussions#3972] [Everyday usage] [answered]
Hi, I'm using homebrew on an M1 Mac

Whenever I install a Perl module that deploys a script (e.g. App::ack comes with ack), they wind up in /opt/homebrew/Cellar/perl/5.34.0/bin. This directory is not in my path. How should I add it in a proper way so as not to break homebrew? I could place it in .zprofile. Is this the appropriate way to do it? I'm wary about updating PATHs in such a way that cause future issues or conflicts between Homebrew and the system versions of software

**Comment by carlocab** [comment, marked as answer]
You can add `$(brew --prefix perl)/bin` to your `PATH`. In your case, that should evaluate to `/opt/homebrew/opt/perl/bin`, which will prevent you from relying on a path that changes when Perl's version changes.

However, I suggest finding a way to make sure those scripts don't end up there. Putting them there will result in their deletion when Perl is upgraded.

**Reply by dhorne** [reply]
Thanks. As you point out, there are likely to be challenges when I switch to a newer version of Perl.  I think I'll switch to perlbrew and keep my Perl distributions separate from both the OS and Homebrew
